### ISSUE TITLE
Enums + jit still not working in PyTorch 2.1 w/ Python 3.11

### ISSUE THREAD

**Issue by rwightman** [pytorch/pytorch#108933]
### 🐛 Describe the bug

I reported this issue a few months back on torch nightlies where torchscript was having issues with a number of `timm` models when using Python 3.11. It seemed like there was an awareness of the issues, plan to fix discussing things in slack, but issue persists in the 2.1 RC..

It looks like it's a problem with an enum used, so any model that has an enum in timm is failing torchscript tests in python 3.11. No problems with any recent pytorch versions in python < 3.11.

``` 
  model = torch.jit.script(model)
            ^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/_script.py", line 1324, in script
    return torch.jit._recursive.create_script_module(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/_recursive.py", line 556, in create_script_module
    concrete_type = get_module_concrete_type(nn_module, share_types)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/_recursive.py", line 505, in get_module_concrete_type
    concrete_type = concrete_type_store.get_or_create_concrete_type(nn_module)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/_recursive.py", line 438, in get_or_create_concrete_type
    concrete_type_builder = infer_concrete_type_builder(nn_module)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/_recursive.py", line 284, in infer_concrete_type_builder
    sub_concrete_type = get_module_concrete_type(item, share_types)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/_recursive.py", line 505, in get_module_concrete_type
    concrete_type = concrete_type_store.get_or_create_concrete_type(nn_module)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/_recursive.py", line 438, in get_or_create_concrete_type
    concrete_type_builder = infer_concrete_type_builder(nn_module)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/_recursive.py", line 395, in infer_concrete_type_builder
    attr_type, inferred = infer_type(name, value)
                          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/_recursive.py", line 226, in infer_type
    ann_to_type = torch.jit.annotations.ann_to_type(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/annotations.py", line 508, in ann_to_type
    the_type = try_ann_to_type(ann, loc, rcb)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/annotations.py", line 489, in try_ann_to_type
    scripted_class = torch.jit._script._recursive_compile_class(ann, loc)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/jit/_script.py", line 1506, in _recursive_compile_class
    rcb = _jit_internal.createResolutionCallbackForClassMethods(obj)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/_jit_internal.py", line 457, in createResolutionCallbackForClassMethods
    captures.update(get_closure(fn))
                    ^^^^^^^^^^^^^^^
  File "/blah/.conda/envs/pytorch-210n/lib/python3.11/site-packages/torch/_jit_internal.py", line 207, in get_closure
    captures.update(fn.__globals__)
                    ^^^^^^^^^^^^^^
AttributeError: 'wrapper_descriptor' object has no attribute '__globals__'
```

### Versions

timm is failing torchscript tests in python 3.11. No problems with any recent pytorch versions in python < 3.11.

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

**Comment by rwightman** [comment]
FYI to reproduce, w/ a newer (0.9.x) `timm` release:
```
import torch
import timm
mm = timm.create_model('vit_base_patch16_224', pretrained=True)
torch.jit.script(mm)
```

**Comment by malfet** [comment]
Original PR that I though should unblock enums for 3.11: https://github.com/pytorch/pytorch/pull/91805

**Comment by malfet** [comment]
Was fixed in lastest 2.1.0 release candidate, tested on my M1 mac by running the following:
```
% python -c "import sys;import torch;import timm;print(sys.version, torch.__version__, torch.version.git_version);torch.jit.script(timm.create_model('vit_base_patch16_224', pretrained=True))"
3.11.5 | packaged by conda-forge | (main, Aug 27 2023, 03:33:12) [Clang 15.0.7 ] 2.1.0 7bcf7da3a268b435777fe87c7794c382f444e86d
```

**Comment by rwightman** [comment]
@malfet thanks, tested in timm with latest RC and looks good
